nuxeo-coldstorage-package/src/main/resources/package.xml

```diff
@@ -7,8 +7,7 @@
   <home-page>https://doc.nuxeo.com/coldstorage/</home-page>
   <vendor>Nuxeo</vendor>
   <platforms>
-    <platform>server-@VERSION@</platform>
-    <platform>server-@VERSION@-*</platform>
+    <platform>server-@NUXEO_TARGET_VERSION@</platform>
   </platforms>
```

Comment:
On this file, we should think if we want to declare dependencies (WebUI as a mandatory/optional?)
Same for the other addons, since all of them don't work without Web UI.

Reply:
I see your point of view, but since if our addons we expose `operation` and `Rest API` I don't know if we have the case where a customer will use only these addons as a backend part and he has its own UI or application and he has no need to add web-ui.

Anyway on the others addons:
- Aspera it's specified as needed package in `package.xml`
- Retention it's not done the same way, IIRC you added the contribution as required in one of the xml contribution

I will update Cold Storage in the sam way that Aspera, but I have a question for you, for retention adding the required on web contribution is it a requirement to make thing work or adding it as a package is a requirement

thx

@nmpcunha

Reply:
Agree. Some of our addons may only be used through REST API. And, in that case we can have the `<optional-dependency>` tag, IIRC.

But, since this is about packaging and we are packaging everything (back end and front end contributions) and people will get everything when installing the package, it is probably sensible to state that `nuxeo-web-ui` is a dependency, otherwise some parts of the installations may even fail. Example: https://github.com/nuxeo/nuxeo-coldstorage/blob/master/nuxeo-coldstorage-web/src/main/resources/OSGI-INF/deployment-fragment.xml

About Retention, guess you are referring to the same contribution as above, right? This will be the expected way of having UI contributions outside Web UI (at least, until we live inside Java Nuxeo world).
So, yes, bottom line is: all these addons (Aspera, Retention, Cold Storage) will have that dependency and will need to state that `nuxeo-web-ui` is a dependency on `package.xml` so it can appear on _Connect_ and also to guide the installation.